### DISCUSSION TITLE
How can I use `Panel.fit` (from rich) instead of the `Panel` default in both the help message and Typer App argument errors?

### DISCUSSION THREAD

**Discussion by Apocryphon-X** [fastapi/typer#607] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer


def main(name: str, lastname: str):
    print(f"Hello {name} {lastname}")


if __name__ == "__main__":
    typer.run(main)
```

### Description

I want to use `Panel.fit` instead of the `Panel` default on help messages and argument error messages.

Default behaviour:
![image](https://github.com/tiangolo/typer/assets/40130428/b95e5967-2de3-40c2-8d56-31ee78a71324)

Example of the behaviour I want to achieve:
![image](https://github.com/tiangolo/typer/assets/40130428/61a86db8-6bcd-4d37-9eac-187d4c7747e1)

P.D: To obtain the second image i used the following code:

```py
import typer
from rich.panel import Panel


def main(name: str, lastname: str):
    print(f"Hello {name} {lastname}")


if __name__ == "__main__":
    typer.rich_utils.Panel = Panel.fit
    typer.run(main)
```

However, this only affects error panels and the help message remains the same. How can I do this for both types of messages?

### Operating System

Linux

### Typer Version

0.9.0

### Python Version

Python 3.11.3

**Comment by harkabeeparolus** [comment]
What a good suggestion! I agree that the error message looks much better with `Panel.fit`.

**Comment by Apocryphon-X** [comment]
Bump

**Comment by con-f-use** [comment]
I'd also like this very much, see https://github.com/tiangolo/typer/discussions/680

Setting `TERMINAL_WIDTH` at least helps to limit it to something sane like `MAX_WIDTH = 200` and corresponds to `rich.console.Console(width=...)

See:
  https://github.com/tiangolo/typer/blob/master/typer/rich_utils.py#L141
  
However, setting [`expand = False`](https://github.com/tiangolo/typer/blob/master/typer/rich_utils.py#L450) in the Panel settings, does exactly nothing, and I'm not sure why.

**Comment by harkabeeparolus** [comment]
I figured it out. 😁

The reason `Panel.fit()` does not work is the use of **Columns** in `_get_parameter_help()`. If you replace Columns with a one-row `Table.grid()`, it works.

Now "all" we need to do is make a PR to expose it through the Typer API and make it configurable for the end user. 😅

**Reply by harkabeeparolus** [reply]
Update: I made a draft PR. #1096

**Comment by sebastianhaberey** [comment]
Yes, Panel.fit() would be nice, or a `max width` of some kind.

Edit: Thinking about this still. I wonder how other projects deal with this? Do people not work with their terminals in fullscreen? Or maybe project maintainers just don't care that it looks weird? Anyways, I think it would be great to get @harkabeeparolus PR merged. I've had a look at it and it doesn't seem that intrusive - should be easy to review for someone with knowledge of the _typer_ codebase.

**Reply by harkabeeparolus** [reply]
It did seem to work fine, but I had two problems:

* I'm not familiar enough with the codebase to write the required tests.

* I'm not sure how much documentation I have to update, or how to take screenshots that look consistent with the rest of the docs.

**Reply by harkabeeparolus** [reply]
I updated my PR #1096 and fixed a bug. Still need help with writing tests, but other than that I believe it is basically ready to merge. 😊
